Source/ProgressionSystemRuntime/Private/LevelActors/PSStarActor.cpp

```diff
@@ -61,6 +88,21 @@ void APSStarActor::OnAnyCinematicStarted(const UObject* LevelSequence, const UOb
 	TryPlayHideStarAnimation();
 }
 
+// Called to initialize the Star menu animation
+void APSStarActor::InitStarMenuAnimation()
+{
+	if (UMyBlueprintFunctionLibrary::GetMyGameState()->GetCurrentGameState() == ECurrentGameState::Menu)
```

Comment:
Never do -> anywhere without checks. GameState might be null is it's regular actor.
- Here, in Init, check it with if (ensureMsgf...) 
- in tick, just do regular if without ensure, as its ticking so will become valid few frames later

Reply:
done as part of https://github.com/h4rdmol/ProgressionSystem/pull/14/commits/fd5af6179b1f774ca9ac43ebdd623c568cee7ff9

Reply:
@h4rdmol, is not addressed yet
```
if (ensureMsgf(UMyBlueprintFunctionLibrary::GetMyGameState()->GetCurrentGameState()
```
In this code, you still might have crash. Just use `AMyGameStateBase::GetCurrentGameState()` that is static, so you don't even need to call GetMyGameState(), see examples in other code by searching `GetCurrentGameState()`

Reply:
done as part of: https://github.com/h4rdmol/ProgressionSystem/pull/14/commits/6141759e381d2b8658971efac878aaebce1a9cf7

Reply:
@h4rdmol is still wrong, just call static function directly:
```
AMyGameStateBase::GetCurrentGameState() == ECGS::Menu
```

Reply:
done https://github.com/h4rdmol/ProgressionSystem/pull/14/commits/9c7ee7a8960dc8cbac8a37fd197a905bc47d19c8